isis/src/tgo/apps/tgocassisrdrgen/main.cpp

```diff
@@ -191,6 +191,31 @@ void IsisMain() {
     focalLengthNode.removeAttribute("unit");
   }
 
+  // Fix the footprint section's output structure
+  QStringList xmlPath2;
+  xmlPath2 << "Observation_Area"
+          << "Discipline_Area"
+          << "geom:Geometry"
+          << "geom:Geometry_Orbiter"
+          << "geom:Surface_Geometry"
+          << "geom:Surface_Geometry_Specific"
+          << "geom:Footprint_Vertices"
+          << "geom:Pixel_Intercept";
+
+  QDomElement pixelInterceptNode = process.getElement(xmlPath2, observationNode);
+  QDomElement pixelInterceptNode2 = pixelInterceptNode.nextSiblingElement();
+  QDomElement pixelInterceptNode3 = pixelInterceptNode2.nextSiblingElement();
+  QDomElement pixelInterceptNode4 = pixelInterceptNode3.nextSiblingElement();
+
+  pixelInterceptNode.insertAfter(pixelInterceptNode.firstChildElement("geom:pixel_latitude"), pixelInterceptNode.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode2.insertAfter(pixelInterceptNode2.firstChildElement("geom:pixel_latitude"), pixelInterceptNode2.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode3.insertAfter(pixelInterceptNode3.firstChildElement("geom:pixel_latitude"), pixelInterceptNode3.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode4.insertAfter(pixelInterceptNode4.firstChildElement("geom:pixel_latitude"), pixelInterceptNode4.firstChildElement("geom:reference_pixel_location"));
+  
+  pixelInterceptNode2.setTagName("geom:Pixel_Intercept");
+  pixelInterceptNode3.setTagName("geom:Pixel_Intercept");
+  pixelInterceptNode4.setTagName("geom:Pixel_Intercept");
```

Comment:
Why is this needed? I'd expect the translation file to put these in the correct places based on the siblings.

Reply:
@jessemapel This is a work-around because I couldn't get it working just via the translation files. I know you worked on the original code for ProcessExportPds4 (etc) as well, do you remember if there's a better way? 

When I just had:

```
Group = Corner1Lattiude
  Auto
  Optional
  InputKey        = Corner1Latitude
  InputPosition   = (IsisCube, Archive)
  OutputSiblings =  ("geom:reference_pixel_location|Lower Left Corner")
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  Translation     = (*, *)
End_Group

Group = Corner1Longitude
  Auto
  Optional
  InputKey        = Corner1Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  Translation     = (*, *)
End_Group

Group = Corner2Lattiude
  Auto
  Optional
  InputKey        = Corner2Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings = ("geom:reference_pixel_location|Lower Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner2Longitude
  Auto
  Optional
  InputKey        = Corner2Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings = ("geom:reference_pixel_location|Lower Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner3Lattiude
  Auto
  Optional
  InputKey        = Corner3Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Left Corner")
  Translation     = (*, *)
End_Group

Group = Corner3Longitude
  Auto
  Optional
  InputKey        = Corner3Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Left Corner")
  Translation     = (*, *)
End_Group

Group = Corner4Lattiude
  Auto
  Optional
  InputKey        = Corner4Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner4Longitude
  Auto
  Optional
  InputKey        = Corner4Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Right Corner")
  Translation     = (*, *)
End_Group
``` 

I got the following: 

.
.
.
```
 <geom:Geometry>
    <geom:Geometry_Orbiter>
     <geom:Surface_Geometry>
      <geom:Surface_Geometry_Specific>
       <geom:Footprint_Vertices>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Lower Left Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">10</geom:pixel_latitude>
         <geom:pixel_longitude unit="deg">1</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">2</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">3</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">4</geom:pixel_longitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Lower Right Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">12</geom:pixel_latitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Upper Left Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">13</geom:pixel_latitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Upper Right Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">14</geom:pixel_latitude>
        </geom:Pixel_Intercept>
       </geom:Footprint_Vertices>
      </geom:Surface_Geometry_Specific>
     </geom:Surface_Geometry>
    </geom:Geometry_Orbiter>
   </geom:Geometry>
```

Reply:
Strange, I don't know why that's happening. It looks like it should spit things out in the expected places. Considering the isisexport work being done, I think this is okay.